frontend/pages/tickets/[[...slug]].tsx

```diff
@@ -85,123 +53,160 @@ const Text = styled.h1`
   font-size: 1rem;
   margin: 0.5rem 0rem;
 `
-// TODO: Add auth handling gracefully.
 
-const Ticket = ({ tickets, buyers, event, home }): ReactElement => {
-  // const Ticket = ({ event }): ReactElement => {
+type TicketProps = InferGetServerSidePropsType<typeof getServerSideProps>
+
+// TODO: Add auth handling gracefully.
 
+const Ticket: React.FC<TicketProps> = ({
+  tickets,
+  buyers,
+  event,
+  home,
+}): ReactElement => {
   if (home) {
     return (
       <Center>
         Welcome to Ticketing! Please browse events with available tickets{' '}
         <a href="/events">here</a>.
       </Center>
     )
-  } else if (!tickets.totals) {
+  } else if (!tickets || !tickets.totals || !tickets.available) {
     return <Center>No tickets found with given user permissions.</Center>
   }
   const { totals, available } = tickets
-  const ticks = {}
-
-  // totals.forEach((tick) => {
-  //   if (ticks[tick.type] == null) {
-  //     ticks[tick.type] = { type: tick.type }
-  //   }
-  //   ticks[tick.type].total = tick.count
-  // })
-
-  // available.forEach((tick) => {
-  //   if (ticks[tick.type] == null) {
-  //     ticks[tick.type] = { type: tick.type }
-  //   }
-  //   ticks[tick.type].available = tick.count
-  // })
-
-  // buyers.forEach((tick) => {
-  //   if (ticks[tick.type] == null) {
-  //     ticks[tick.type] = { type: tick.type }
-  //   }
-  //   if (ticks[tick.type].buyers == null) {
-  //     ticks[tick.type].buyers = []
-  //   }
-
-  //   ticks[tick.type].buyers.push(tick.fullname)
-  // })
-
-  // tickets = []
-  // for (const [_, value] of Object.entries(ticks)) {
-  //   tickets.push(value)
-  // }
+  const tickTypes = {}
+
+  // For given ticket type, get the total and available tickets, as well as buyers
+  for (const ticket of totals) {
```

Comment:
Duplicate entry of ticket.type and map key

Reply:
Not ideal but kept since map is more appropriate for the (hacky) way we populate tickTypes but still need the type later on, so avoids the inevitable intermediate step of feeding the key back into the value.